agent/api/task.go

```diff
@@ -462,6 +505,11 @@ func (task *Task) UpdateDesiredStatus() {
 	task.updateContainerDesiredStatus()
 }
 
+func (task *Task) SetKnownStatus(status TaskStatus) {
+	task.KnownStatus = status
```

Comment:
Can we just use the two functions: setKnownStatus and updateKnownStatusTime which has been defined below.